### PR TITLE
Non repeatable number task

### PR DESCRIPTION
**Pull request by yest89**
Non repeatable number task

### CODE REVIEW

File: task31/task31.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include <iostream>
+#include <vector>
+#include <map>
+#include <algorithm>
+
+/*
+��� ������ ������� ����, � ������� � ������ ����� ������� ��������
+����� VS ��������� �����, ����� ����� ������� �����, � ������� ������ ����������.
+����� ����� ������� ��� ������� (PR) � ������ ���.
+*/
```
Comment (ak-toktor) [inline task31/task31.cpp]: Russian comments

File: task31/task31.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        }
+
+        if (repeatableFlag == 1) {
+            foundNonRepeatableNumber = nums[i];
+            break;
+        }
+    }
+    if (repeatableFlag == false) {
+        throw std::runtime_error("There is no non repeatable number");
```
Comment (ak-toktor) [inline task31/task31.cpp]: it is not the best way to throw exception in such case. because it is normal situation

File: task31/task31.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+/*
+��� ������ ������� ����, � ������� � ������ ����� ������� ��������
+����� VS ��������� �����, ����� ����� ������� �����, � ������� ������ ����������.
+����� ����� ������� ��� ������� (PR) � ������ ���.
+*/
+
+
+int findNonRepeatableNumber(std::vector<int>& nums) {
+    bool repeatableFlag = false;
+    int foundNonRepeatableNumber = -999;
```
Comment (ak-toktor) [inline task31/task31.cpp]: -999 - something strange )

File: task31/task31.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+int main()
+{
+    std::vector<int> numbers{ 7, 9, 1, 7, 8, 0, 9, 1, 8 };
+    std::vector<int> numbersWithoutRepeatableNumber { 7, 9, 1, 7, 8, 9, 1, 8 };
+
+    try {
+        std::cout << "Non repeatable number: " << findNonRepeatableNumber(numbers) << std::endl;
+    }
+    catch (std::exception& ex) {
```
Comment (ak-toktor) [inline task31/task31.cpp]: catch exceptions as a const reference

**Review by ak-toktor — CHANGES_REQUESTED** [review]

File: task31/task31.cpp (hunk 3 above)
Comment (yest89) [reply task31/task31.cpp]: done. thank you

File: task31/task31.cpp (hunk 2 above)
Comment (yest89) [reply task31/task31.cpp]: i agree. done. thank you

File: task31/task31.cpp (hunk 1 above)
Comment (yest89) [reply task31/task31.cpp]: done. thank you

File: task31/task31.cpp (hunk 4 above)
Comment (yest89) [reply task31/task31.cpp]: done. thank you

**Pull request merged**